cmd/katib-controller/v1beta1/main.go

```diff
@@ -108,6 +109,10 @@ func main() {
 		HealthProbeBindAddress: healthzAddr,
 		LeaderElection:         enableLeaderElection,
 		LeaderElectionID:       leaderElectionID,
+		// TODO: Once the below issue is resolved, we need to switch discovery-client to the built-in one.
+		// https://github.com/kubernetes-sigs/controller-runtime/issues/2354
+		// https://github.com/kubernetes-sigs/controller-runtime/issues/2424
+		MapperProvider: apiutil.NewDiscoveryRESTMapper,
```

Comment:
Is MapperProvider mandatory for controller-runtime >= 0.15 ?

Reply:
@tenzen-y This error occurs only in unit test during env-test run or in actual Kubernetes cluster as well ?

Reply:
The root cause is that even though `meta.IsNoMatchError` checks if the error is `No{Kind,Resource}MatchError`, the controller runtime v0.15.0 uses `ErrGroupDiscoveryFailed` instead of `No{Kind,Resource}MatchError` as a NoMatching error. 

So, this error will occur anywhere.